### PR TITLE
Delete unused global variables from mysqld.h

### PR DESCRIPTION
**Pull request by robinnewhouse**
## Description
Several variables declared in mysqld.h appear to be old system variables
that have been left over after deprecation. Delete them using IDE
refactoring to automatically search for other uses. Most cases had no
other uses in the code.

slave_allow_batching had a test that was effectively unused, as the
result was only
`-ERROR HY000: Unknown system variable 'slave_allow_batching'`
so that was deleted as well.

Build and test still works without issue as expected.

Examples of variables that were deleted and not cleaned up:

* 0c571f8 removed the last usage of `VERS_ALTER_HISTORY_KEEP` but did not delete it from the enum.
* 7c58e97 moved `key_memory_quick_index_merge_root` (and similar) to `mysqld.cc` to `mysqld.h`, but still was not used anywhere.
* 23d8586 moved `key_thread_kill_server` along with others to header file, but still was used nowhere. I can't actually find where it was ever used except being defined in a header.

## Release Notes
None

## How can this PR be tested?

Building the server and running the MTR test suite shows no failures.

## Basing the PR against the correct MariaDB version
- [x] *This is a bug fix and the PR is based against the earliest maintained branch in which the bug can be reproduced.*

Note: As discussed with @LinuxJedi, this is targeting the earliest maintained branch to ease merging upwards, even though it is not necessarily a bugfix.

## PR quality check
- [x] I checked the [CODING_STANDARDS.md](https://github.com/MariaDB/server/blob/-/CODING_STANDARDS.md) file and my PR conforms to this where appropriate.
- [x] For any trivial modifications to the PR, I am ok with the reviewer making the changes themselves.

### CODE REVIEW

**Comment by vaintroub** [comment]
A general suggestion : can we do pure refactoring task in current development branch, i.e rebase the patch onto 11.6?
In the old versions, we usually only fix bug. Large refactoring can make a merge problematic, and  between 10.5 and 11.6, there are many merges.

**Comment by robinnewhouse** [comment]
> A general suggestion : can we do pure refactoring task in current development branch, i.e rebase the patch onto 11.6? In the old versions, we usually only fix bug. Large refactoring can make a merge problematic, and between 10.5 and 11.6, there are many merges.

I had debated on which branch to target this. Originally it was targeting 11.5 (latest at the time) for exactly this reason. @LinuxJedi had suggested targeting an earlier branch so that later modifications to earlier branches (say a 10.6 bugfix) would face fewer merge conflicts when merging upwards.

I am happy to switch back to targeting the latest development branch if that is what is preferred though.

**Comment by vaintroub** [comment]
> > A general suggestion : can we do pure refactoring task in current development branch, i.e rebase the patch onto 11.6? In the old versions, we usually only fix bug. Large refactoring can make a merge problematic, and between 10.5 and 11.6, there are many merges.
> 
> I had debated on which branch to target this. Originally it was targeting 11.5 (latest at the time) for exactly this reason. @LinuxJedi had suggested targeting an earlier branch so that later modifications to earlier branches (say a 10.6 bugfix) would face fewer merge conflicts when merging upwards.
> 
> I am happy to switch back to targeting the latest development branch if that is what is preferred though.

I asked around, and the consensus is refactoring/cosmetics go into newest branch, as to avoid accidental mistakes when mergin g -  it already happened that seemingly harmless refactoring causes bugs when merged (I do not think your work would cause something like that, but this is just a guideline). So, refactoring would handled more like new feature, goes into development branch, and unlike bugfixing, which goes into earliest applicable branch. @LinuxJedi agreed to change [contribution guideline](https://mariadb.org/get-involved/getting-started-for-developers/submitting-pull-request/) to explicitly mention refactoring, because currently it only talks about bugs vs new features.

With that, can you maybe base your work on 11.6?

File: sql/mysqld.cc [hunk 1]
```diff
@@ -339,7 +339,6 @@ static char *character_set_filesystem_name;
 static char *lc_messages;
 static char *lc_time_names_name;
 char *my_bind_addr_str;
-int server_socket_ai_family;
```
[the remark this answers is not in the document]
Comment (robinnewhouse) [reply sql/mysqld.cc]: With the rebase onto 11.6, this is no longer an issue. This is already gone in the 11.6 branch.

**Comment by vaintroub** [comment]
Thank you! Looks mostly good.  Can we do pure whitespace fixes in a separate commit (if at all)?

**Comment by robinnewhouse** [comment]
> Thank you! Looks mostly good. Can we do pure whitespace fixes in a separate commit (if at all)?

Yes I considered that. I'll remove the pure whitespace changes.

**Review by vaintroub — APPROVED** [review]
Looks good to me

**Pull request merged**